packages/xo-server-load-balancer/src/performance-plan.js

```diff
@@ -45,25 +45,27 @@ export default class PerformancePlan extends Plan {
       toOptimizeOnly: true,
     })
 
-    if (!results) {
-      return
+    if (results) {
+      const { averages, toOptimize } = results
+      toOptimize.sort((a, b) => -this._sortHosts(a, b))
+      for (const exceededHost of toOptimize) {
+        const { id } = exceededHost
+
+        debug(`Try to optimize Host (${exceededHost.id}).`)
+        const availableHosts = filter(hosts, host => host.id !== id)
+        debug(`Available destinations: ${availableHosts.map(host => host.id)}.`)
+
+        // Search bests combinations for the worst host.
+        await this._optimize({
+          exceededHost,
+          hosts: availableHosts,
+          hostsAverages: averages,
+        })
+      }
     }
 
-    const { averages, toOptimize } = results
-    toOptimize.sort((a, b) => -this._sortHosts(a, b))
-    for (const exceededHost of toOptimize) {
-      const { id } = exceededHost
-
-      debug(`Try to optimize Host (${exceededHost.id}).`)
-      const availableHosts = filter(hosts, host => host.id !== id)
-      debug(`Available destinations: ${availableHosts.map(host => host.id)}.`)
-
-      // Search bests combinations for the worst host.
-      await this._optimize({
-        exceededHost,
-        hosts: availableHosts,
-        hostsAverages: averages,
-      })
+    if (this._balanceVcpu) {
```

Comment:
You are right, the vCPU balancing can't cause hosts to get overloaded, but currently it can trigger the migration of VMs that just migrated because of performance reasons. Maybe we should add a condition to test if we should execute the vCPU prepositionning or not, but for the moment I don't know what this condition should be.

Reply:
Maybe we should measure the global average CPU load, and only execute the vCPU prepositionning if that value is under a fixed threshold ?
It would not totally avoid the situation we are discussing, but it should make it way less frequent.